db/write_thread.h

```diff
@@ -185,6 +185,7 @@ class WriteThread {
     CommitRequest* request;
     SequenceNumber sequence;  // the sequence number to use for the first key
     Status status;
+    std::mutex status_mu;
```

Comment:
You should figure out another solution. Each write creates a writer, this puts too much load on futex infrastructure (last I checked each futex creation insert to a global kernel map).